content/en/developers/dogstatsd/high_throughput.md

```diff
@@ -218,6 +218,140 @@ Then set the Agent `dogstatsd_so_rcvbuf` configuration option to the same number
 dogstatsd_so_rcvbuf: 26214400
 ```
 
+### Over UDS (Unix Domain Socket)
+
+#### Linux
+
+Some Linux distributions have a kernel datagram queue size which can be low. This can be confirmed using the following command:
+
+```bash
+sysctl net.unix.max_dgram_qlen
+```
+
+If the value is < 512, you can increase it to 512 or more using this command:
+
+```bash
+sysctl -w net.unix.max_dgram_qlen=512
+```
+
+Add the following configuration to `/etc/sysctl.conf` to make this change permanent:
+
+```conf
+net.unix.max_dgram_qlen = 512
+```
+
+In the same manner, the `net.core.wmem_max` could be incremented to 25MiB to improve
+client writing performances:
+
+```conf
+net.core.wmem_max = 26214400
+```
+
+## Configuration optimization guide
```

Comment:
Most of this section probably shouldn't live in our public docs:

* I think that the config options mentioned here should not be considered part of the "officially-supported config options", so that we have room to break them in the future as we improve performance and perf options in the daemon. But let me know what you think of course.
* overall this seems a bit too complex for end users, and goes into lots of internal details that are very interesting to us but not really to the end user.

For now, in the public docs here, I think we should stick to general recommendations like:

* recommend to use the latest version of an official dogstatsd client (which should have best performance and by default use the right defaults for UDS and UDP datagram sizes -- if it's not the case now for some clients it will be in a couple of months)
* mention these "optimal" datagram sizes for users using non-official clients.
* if we add a setting that sets at a high level in the Agent the sizes of the buffers, only present here this single setting, and describe what the tradeoff is